File: project/.sdk/tm/py/utility/transform_request.py
# ProjectName SDK utility: transform_request

from __future__ import annotations
from utility.voxgig_struct import voxgig_struct as vs
from core.helpers import to_map


def transform_request_util(ctx):
    spec = ctx.spec
    point = ctx.point

    if spec is not None:
        spec.step = "reqform"

    transform = to_map(vs.getprop(point, "transform"))
    if transform is None:
        return ctx.reqdata

    reqform = vs.getprop(transform, "req")
    if reqform is None:
        return ctx.reqdata

    reqdata = vs.transform({
        "reqdata": ctx.reqdata,
    }, reqform)

    return reqdata

File: project/.sdk/tm/rb/utility/transform_request.rb
# ProjectName SDK utility: transform_request
require_relative 'struct/voxgig_struct'
require_relative '../core/helpers'
module ProjectNameUtilities
  TransformRequest = ->(ctx) {
    spec = ctx.spec
    point = ctx.point
    spec.step = "reqform" if spec
    transform = ProjectNameHelpers.to_map(VoxgigStruct.getprop(point, "transform"))
    return ctx.reqdata unless transform
    reqform = VoxgigStruct.getprop(transform, "req")
    return ctx.reqdata unless reqform
    VoxgigStruct.transform({ "reqdata" => ctx.reqdata }, reqform)
  }
end
